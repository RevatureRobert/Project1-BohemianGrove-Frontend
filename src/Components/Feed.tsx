<<<<<<< HEAD
import React from 'react';
import CardContainer from '../Components/Card';
=======
import React, { useState } from 'react';
import CardContainer from './Card';
>>>>>>> mohamed-hassan
import { Component } from 'react';
import { Card, ListGroup, ListGroupItem, 
        CardBody, CardImg, CardTitle, 
        CardSubtitle, CardText, FormGroup,
        Label, Input, Button } from 'reactstrap';
import { render } from '@testing-library/react';
import '../Styles/Card.css';
import thumbnail from '../thumbnails-z1-1.png';
import { string } from 'yargs';
<<<<<<< HEAD
=======
import axios from 'axios';
import { connect } from 'react-redux';

>>>>>>> mohamed-hassan


class Feed extends Component<any, any> {

<<<<<<< HEAD
    constructor(props){
        super(props);
       
        this.state = {
          postsList:[],
          newInput: ' ',
        };
      }

    //   componentDidCatch() {
    //       console.error();
          
    //   }

    componentDidMount() {
        //TODO
        //Fetch posts from endpoint and pass to state
    
        
        let posts = [
            {
                username: "John Smith",
                handle: "JohnnyAppleSeed",
                timestamp: "12.23.12:11.22.34",
                img: "thumbnail",
                body: "Peter Parker Piper Pan",
            },
            {
                username: "Jane Doe",
                handle: "GrabAPailOfWater",
                timestamp: "11.37.23:44.48.09",
                img: "thumbnail",
                body: "Some random shitpost.",
            }
        ]
        this.setState( {postsList : posts});
        
    }

    // addNewPost() {
    //     //TODO
    //     //Add new post via POST endpoint
    // }

    renderItems()  {
        return ( this.state.postsList.map((card) => (
            <div id="CardContainer">
            <Card >              
              <CardBody>
              <CardImg id ="card-image" top width="10px" height = "80px"  src={thumbnail} alt="Card image cap" />
                <CardTitle id="card-title" tag="h5">{card.handle}</CardTitle>
                <CardSubtitle id="card-subtitle" tag="h6" className="mb-2 text-muted">@{card.username}</CardSubtitle>
                
                
              </CardBody>
              <CardText id="body">{card.body}</CardText>
              <CardText id="timestamp">{card.timestamp}</CardText>
            </Card>
          </div>

            ))
        );
        }

       

       

    render() {
        return(
            // onChange={(event) => this.updateNewInput(event.target.value) }
            <div >
=======
    // const[postsList, setPostsList] = useState([]);
    // const[newInput, setNewInput] = useState(null);


    constructor(props){
        super(props);
        this.state = {
          postsList:[],
          newInput: ' ',
          loginToken: ' '
        };
      }

      

    async componentDidUpdate() {
        let posts: string[] = [];
        //this.setState({ postsList : initialState.user});
        const res =  await axios.get('http://localhost:3000/api/posts').then(res => {
            console.log(res.data);
            res.data.forEach(post => posts.push(post)); 
            posts.reverse();
            this.setState({ postsList : posts });
        });
      }

    async componentDidMount() {
        //TODO
        //Fetch posts from endpoint and pass to state
        let posts: string[] = [];
        //this.setState({ postsList : initialState.user});
        const res =  await axios.get('http://localhost:3000/api/posts').then(res => {
            console.log(res.data);
            res.data.forEach(post => posts.push(post)); 
            this.setState({ postsList : posts });
        });

        
    }


    createNewDate(time) {
        return new Date(time).toLocaleString();
    }

    renderItems()  {
        return this.state.postsList.map((card) => (
            <div id="CardContainer">
            <Card >              
              <CardBody>
              <CardImg id ="card-image" top width="10px" height = "80px"  src={thumbnail} alt={thumbnail} />
                <CardTitle id="card-title" tag="h5">{card.displayName}</CardTitle>
                <CardSubtitle id="card-subtitle" tag="h6" className="mb-2 text-muted">@{card.userName}</CardSubtitle>
                
                
              </CardBody>
              <CardText id="body">{card.postBody}</CardText>
              <CardText id="timestamp">{ this.createNewDate(card.postTime)}</CardText>
            </Card>
          </div>

            ));
        }

        updatePostState() {
            //this.setState( {postsList :posts});
        }

        updateNewInput(e) {
            this.setState({ newInput: e });
            
        }

        createPost() {
            //TODO
            //Get user credentials from token
            //Send Post request to server
            let currTime = new Date().toLocaleString();
            const postBody = { post: { userName: "JohnCena", displayName: 'NowYouSeeMeNowYouDont', displayImg: 'undefined',
                 postBody: this.state.newInput, postImg: 'undefined'}};

            axios.post('http://localhost:3000/api/posts', postBody).then(resp => console.log(resp));
        }

    render() {
        return(
            <div key={this.state.index}>
>>>>>>> mohamed-hassan

                <Card id="AddPostContainer">
                    <CardBody>
                        <CardTitle id="new-card-title" tag="h5">Add new: </CardTitle>
                        <FormGroup>
                            <Label for="exampleText">What's on your mind?</Label>
<<<<<<< HEAD
                            <Input type="textarea" name="text" id="exampleText" />
                        </FormGroup>
                        <Button id="post-button" color="primary">Post</Button>
=======
                            <Input 
                                type="textarea" 
                                name="text" 
                                id="exampleText" 
                                onChange={(e) => this.setState({ newInput: e.target.value })}/>
                        </FormGroup>
                        <Button onClick={() => this.createPost()} id="post-button" color="primary">Post</Button>
>>>>>>> mohamed-hassan
                    </CardBody>
                </Card>
            
                <ListGroup>
                    {this.renderItems()}
                </ListGroup>
    
    
            </div>
        )
    }
}

<<<<<<< HEAD
export default Feed;
=======
export default connect(
    null, {}
) (Feed);
>>>>>>> mohamed-hassan
