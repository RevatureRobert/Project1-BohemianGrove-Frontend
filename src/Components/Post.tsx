import {
    Card,
    CardImg,
    CardText,
    CardBody,
    CardTitle,
    CardSubtitle,
    Button
} from 'reactstrap';

import { IPost } from '../Models/Post'
import { Redirect } from 'react-router-dom';
import { useState } from 'react';
import { useSelector } from 'react-redux';
import { IAppState } from '../Redux/Store';
import axios from 'axios';

interface IProps {
    post: IPost
}

const Post: React.FC<IProps> = (props: IProps) => {
    const user = useSelector((store: IAppState) => store.user);

    const [redirect, setRedirect] = useState('');

    function deletePost(userName: string, postTime: number) {
        axios.delete("http://localhost:3000/api/posts/", {data: {post: {userName, postTime}}}).then(resp => {
            setRedirect("/feed");
        });
    }

    if (redirect) return <Redirect to={redirect} />

    return (
        <div id="CardContainer">
            <Card >
                <CardBody>
                    <CardImg id="card-image" top width="10px" height="80px" src={props.post.displayImg} alt={props.post.displayImg} />
                    <CardTitle style={{ cursor: 'pointer' }} id="card-title" tag="h5" onClick={(e) => setRedirect("/" + props.post.userName)}>{props.post.displayName}</CardTitle>
                    <div className="handleAndButton">
                    {(user?.userName === props.post.userName) && <Button className = "deleteButton" onClick={() => deletePost(user.userName, props.post.postTime || 0)}>X</Button>}
                    <CardSubtitle id="card-subtitle" tag="h6" className="mb-2 text-muted">@{props.post.userName}</CardSubtitle> 
                    </div>
                    </CardBody>
                    <CardText id="body">{props.post.postBody}</CardText>
                    {props.post.postImg && <CardImg src={props.post.postImg} alt={props.post.postImg} />}
                    <CardText id="timestamp">{new Date(props.post.postTime || 0).toLocaleString()}</CardText>
                    
            </Card>
        </div>
    );
}

export default Post;