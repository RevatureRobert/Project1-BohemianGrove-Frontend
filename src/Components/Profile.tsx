import React from "react";
<<<<<<< HEAD
<<<<<<< HEAD
import { Component } from "react";

class ProfileComponent extends Component<any,any> {
=======
=======
>>>>>>> 2981b61005d81a442f7d22030b3890d2489d715d
import { Component, useState } from "react";
import { ListGroup, Card, CardBody, CardImg, Container, CardTitle, CardSubtitle, CardText,
        Nav, NavItem, NavLink, TabContent, Row, Col, TabPane, Button } from 'reactstrap';
import thumbnail from '../thumbnails-z1-1.png';
import '../Styles/Profile.css';
import axios from 'axios';



class ProfileComponent extends Component<any,any> {
    
    constructor(props){
        super(props);
        this.state = {
          postsList:[],
          
        };
      }

    componentDidMount() {
        //TODO
        //Pull user specific posts from endpoint upon initialization of component
        //Also pull user profile
        

    }

    renderProfile() {
        //TODO
        //Pull user profile from endpoint and populate card --> Return card
        return ( <div></div>);
    }

    //Render individual cards for each post
    renderItems() {
        return this.state.postsList.map((card) => (
            <div id="CardContainer">
            <Card >              
              <CardBody>
              <CardImg id ="card-image" top width="10px" height = "80px"  src={thumbnail} alt="Card image cap" />
                <CardTitle id="card-title" tag="h5">{card.handle}</CardTitle>
                <CardSubtitle id="card-subtitle" tag="h6" className="mb-2 text-muted">@{card.username}</CardSubtitle>
                
                
              </CardBody>
              <CardText id="body">{card.body}</CardText>
              <CardText id="timestamp">{card.timestamp}</CardText>
            </Card>
          </div>

            ));
    }
<<<<<<< HEAD
>>>>>>> mohamed-hassan
=======
>>>>>>> 2981b61005d81a442f7d22030b3890d2489d715d

    render() {
        return (
            <div className="globalFeedContainer">

<<<<<<< HEAD
<<<<<<< HEAD
=======
=======
>>>>>>> 2981b61005d81a442f7d22030b3890d2489d715d
            <Row>
                <Col>
                //TODO
                //User Profile Card
                <Container>
                    {this.renderProfile()}
                </Container>
                
                </Col>
                <Col>
                <h4>User History</h4>
                
                <div >
                <Container>
                    <ListGroup>
                        {this.renderItems()}
                    </ListGroup>
                </Container>
                </div>
            
                </Col>

            </Row>
                
<<<<<<< HEAD
>>>>>>> mohamed-hassan
=======
>>>>>>> 2981b61005d81a442f7d22030b3890d2489d715d
            </div>
        )
    }
}

export default ProfileComponent;