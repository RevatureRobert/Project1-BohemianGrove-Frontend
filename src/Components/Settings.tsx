import React from "react";
import { Component } from "react";
import { Row, Container, Col, Form, FormGroup, Card, CardBody, CardSubtitle, CardText, CardTitle, CardImg, Label, Input, Button } from 'reactstrap';
import thumbnail from '../thumbnails-z1-1.png';
import '../Styles/Settings.css'
import axios from 'axios';

class SettingsComponent extends Component<any, any> {

    constructor(props) {
        super(props);
        this.state = {
            username: ' ',
            password: ' ',
            displayname: ' ',
            email: ' ',
            currUser : [],
            validate: {
                emailState: ' '
            }
        }
    }


    //TODO
    //Render form for updating profile information
    componentDidMount() {
        const userName = "kai";
        axios.get('http://localhost:3000/api/users/kai').then(resp => {
            console.log(resp);
            this.setState({ currUser: resp.data.data });
        })
    }

    renderProfileCard() {
        return  (
            <div id="CardContainer">
            <Card >              
              <CardBody>
              <CardImg id ="card-image" top width="10px" height = "80px"  src={thumbnail} alt="Card image cap" />
                <CardTitle id="card-title" tag="h5">{this.state.currUser.displayName}</CardTitle>
                <CardSubtitle id="card-subtitle" tag="h6" className="mb-2 text-muted">@{this.state.currUser.userName}</CardSubtitle>
                                
              </CardBody>
         
            </Card>
          </div>

        )}
        
       
        updateUser() {
            const userObject = { loginToken: " ", user: { userName: "kai", displayName: "Jared", email: "revature@revature.net", profileImg: "undefined"}};
            axios.put('http://localhost:3000/api/users', userObject).then(resp => console.log(resp));
        }

        submitForm(e) {
            //TODO
            //Update user along endpoint http://localhost:3000/api/users PUT
            let eue = e.target.value;
            
        }

        validateEmail(e) {
            const emailRex =
              /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
        
            const { validate } = this.state;
        
            if (emailRex.test(e.target.value)) {
              validate.emailState = 'has-success';
            } else {
              validate.emailState = 'has-danger';
            }
        
            this.setState({ validate });
          }

         componentDidUpdate() {
             //TODO
             //Rerender profile card when user submits update
         }

    render() {
        return(
            <div>
                <Row>
                    <Col>
                        { this.renderProfileCard() }
                    </Col>
                    <Col>
                       
            <div id="update-settings-form">
            <Form className="form" onSubmit={(e) => this.submitForm(e)} >
              <FormGroup>
                <Label for="exampleEmail">Update Settings</Label>                
              </FormGroup>

              <FormGroup>
                  <Label for="username">Username</Label>
                  //TODO --= Change to text field
                  <Input
                  type="text"
                  name="username"
                  id="username"
                  onChange = {(e) => this.setState({ username: e.target.value })}
                  />
              </FormGroup>

              <FormGroup>
                  <Label for="displayName">Handle</Label>
                  <Input
                  type="text"
                  name="handle"
                  id="displayName"
                  onChange = {(e) => this.setState({ displayname: e.target.value })}
                  />
              </FormGroup>
              
              <FormGroup>
                <Label for="examplePassword">Password</Label>
                <Input
                  type="password"
                  name="password"
                  id="examplePassword"
                  placeholder="********"
                  onChange = {(e) => this.setState({ password: e.target.value })}
                />
              </FormGroup>

              <FormGroup>
                <Label for="exampleEmail">Email</Label>
                <Input
                  type="email"
                  name="email"
                  id="exampleEmail"
                  placeholder="teamOne@revature.com"
                  valid={this.state.validate.emailState === "has-success"}
                  invalid={this.state.validate.emailState === "has-danger"}
                  onChange={(e) => {
                    this.validateEmail(e);
                    this.setState({ email: e.target.value });
                  }}
                />
              </FormGroup>

              <Button color="primary" id="submit-button" onClick={() => this.updateUser()} onSubmit={(e) => this.submitForm(e)}>Submit</Button>
            </Form>
            </div>
                    </Col>
                </Row>

            </div>
        )
    }
}

export default SettingsComponent;