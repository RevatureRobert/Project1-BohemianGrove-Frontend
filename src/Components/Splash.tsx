import React from 'react';
import '../Styles/Splash.css';
import {withRouter} from 'react-router'




class SplashComponent extends React.Component<any, any>{

    constructor(props) {
        super(props);
    }

    // signinClick(){
    //     this.props.history.push("/signin");
    // }

    // signupClick()
    // {
    //     this.props.history.push("/signup");
    // }
    

    render() {
        return (
        
        <div id="parent-div">
                <div id ="splash-div" className= "splash-component">
        
                    <h1 id="splash-header">Welcome to Bohemian Grove, the social media site for conspiracy theorists
                    </h1>
                    
                </div> 
    </div>
      );
    }
}
export default withRouter(SplashComponent);


