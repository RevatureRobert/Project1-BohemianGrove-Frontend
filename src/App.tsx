import React from 'react';
import logo from './logo.svg';
import './Styles/App.css';
import SignInComponent from './Components/SignIn';
import SignUpComponent from './Components/SignUp';
import NavigationBar from './Components/NavBar';
import CardContainer from './Components/Card';
import SettingsComponent from './Components/Settings';
import ProfileComponent from './Components/Profile';
import Feed from './Components/Feed';
import { BrowserRouter as Router, Switch, Route, Link } from 'react-router-dom';

import { Component } from 'react';
import {
  Button,
  Form,
  FormGroup,
  Input,
  Label,
  FormFeedback,
  Badge
} from 'reactstrap';

class App extends Component<any, any> {
  render() {
    return (
      
        <div>
          <NavigationBar />
        <Router>
          <Switch>
            <Route path="/signin">
              <SignInComponent />
            </Route>
            <Route path="/signup">
              <SignUpComponent />
            </Route>
            <Route path="/feed">
              <Feed />
            </Route>
            <Route path="/settings">
              <SettingsComponent />
            </Route>
            <Route path="/profile">
              <ProfileComponent />
            </Route>
            
          </Switch>
        </Router>
        </div>
      
    )
  }
}

export default App;


// class App extends Component<any, any> {
//   constructor(props) {
//     super(props);
//     this.state = {
//       email: '',
//       password: '',
//       validate: {
//         emailState: '',
//       },
//     };
//     this.handleChange = this.handleChange.bind(this);
//   }

//   handleChange = (event) => {
//     const { target } = event;
//     const value = target.type === 'checkbox' ? target.checked : target.value;
//     const { name } = target;

//     this.setState({
//       [name]: value,
//     });
//   };

//   validateEmail(e) {
//     const emailRex =
//       /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;

//     const { validate } = this.state;

//     if (emailRex.test(e.target.value)) {
//       validate.emailState = 'has-success';
//     } else {
//       validate.emailState = 'has-danger';
//     }

//     this.setState({ validate });
//   }

//   submitForm(e) {
//     e.preventDefault();
//     console.log(`Email: ${this.state.email}`);
//   }
//   render() {
//     const { email, password } = this.state;
//     return (
//       <div className="App">
//         <h2>Bohemian Grove</h2>
//         <Form className="form" onSubmit={(e) => this.submitForm(e)}>
          

//           <FormGroup>
//                   <Label for="username">Username</Label>
//                   <Input
//                   type="text"
//                   name="username"
//                   id="username"
//                   placeholder="Login"
//                   />
//               </FormGroup>

//           <FormGroup>
//             <Label for="examplePassword">Password</Label>
//             <Input
//               type="password"
//               name="password"
//               id="examplePassword"
//               placeholder="********"
//             />
//           </FormGroup>
//         <Button id="submit-button">Submit</Button>
        
//         <Label href="#" onClick= {() => {<SignUpComponent />} } id="redirect-to-signup" >New User? Start here.</Label>  
        
        
//       </Form>
//     </div>
//   );
// }
// }

// export default App;



