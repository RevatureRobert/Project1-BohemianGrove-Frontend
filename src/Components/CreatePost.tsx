import axios from "axios";
import React, { useEffect } from "react";
import { useState } from "react";
import { useSelector } from "react-redux";
import { Redirect } from "react-router-dom";
import { Card, CardBody, CardTitle, FormGroup, Label, Input, Button } from "reactstrap";
import { IPost } from "../Models/Post";
import { IAppState } from "../Redux/Store";


const CreatePost: React.FC = (props: any) => {
    const user = useSelector((state: IAppState) => state.user);
    
    const [postInfo, setPostInfo] = useState<any>({});
    const [redirect, setRedirect] = useState('');

    function createPost() {
        let post: IPost = {
            userName: user?.userName || '',
            displayName: user?.displayName || '',
            displayImg: user?.profileImg || '',
        }
        if (postInfo.postBody) post = {...post, postBody: postInfo.postBody};
        if (postInfo.postImg) post = {...post, postImg: postInfo.postImg};

        console.log("Sending request...", {post});
        axios.post("http://localhost:3000/api/posts/", {post}).then(resp => {
            setRedirect("/reloadFeed");
        });
    }

    if (redirect) return <Redirect to={redirect}/>;
    
    return (
        <div>
            <Card id="CardContainer">
                <CardBody>
                    <CardTitle id="new-card-title" tag="h5">Add new: </CardTitle>
                    <FormGroup>
                        <Label for="exampleText">What's on your mind?</Label>
                        <Input
                            type="textarea"
                            name="text"
                            id="exampleText"
                            onChange={(e) => setPostInfo({...postInfo, postBody: e.target.value})} />
                    </FormGroup>
                    <Button onClick={() => createPost()} id="post-button" color="primary">Post</Button>
                </CardBody>
            </Card>
        </div>
    );
}

export default CreatePost;