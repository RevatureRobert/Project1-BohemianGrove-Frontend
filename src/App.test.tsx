import React from 'react';
import {render, screen } from '@testing-library/react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import Login from  './Screens/Login';
import SignUp from './Screens/SignUp';
import NavigationBar from './Components/NavBar';
import Settings from './Screens/Settings';
import SettingsComponent from './Components/Settings';
import Profile from './Screens/Profile';
import Feed from './Components/Feed';
import GlobalFeed from './Screens/GlobalFeed';
import Splash from './Screens/Splash';
import Search from './Screens/Search';
import Router from 'react-router-dom';
import App from './App';
import { initialState } from './Redux/Store';
import { mount, shallow } from 'enzyme';
import  Adapter  from '@wojtekmaj/enzyme-adapter-react-17'
import enzyme from 'enzyme'
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { Reducer } from 'react'
import { reducers } from './Redux/Reducers';
import CreatePost from './Components/CreatePost';
import Post from './Components/Post';
import UserList from './Components/UserList';
import User from './Components/User'



const store = createStore(reducers);

enzyme.configure({ adapter: new Adapter() });

describe('SignUpComponent', ()=>{
  it('renders without crashing', () => {
  const component = shallow(<MemoryRouter><SignUp/></MemoryRouter>);
  expect(component).toMatchSnapshot();  
  });
})

describe('SignInUpComponent', ()=>{
  it('renders without crashing', () => {
  const component = shallow(<MemoryRouter><Login/></MemoryRouter>);
  expect(component).toMatchSnapshot();  
  });
})

describe('SettingsComponent', ()=>{
  it('renders without crashing', () => {
  const component = mount(<Provider store={store}><MemoryRouter><Settings/></MemoryRouter></Provider>);
  expect(component).toMatchSnapshot();  
  });
})

describe('GlobalFeedComponent', ()=>{
  it('renders without crashing', () => {
  const component = shallow(<Provider store={store}><MemoryRouter><GlobalFeed/></MemoryRouter></Provider>);
  expect(component).toMatchSnapshot();  
  });
})

describe('ProfileComponent', ()=>{
  it('renders without crashing', () => {
  const component = shallow(<Provider store={store}><MemoryRouter><Profile/></MemoryRouter></Provider>);
  expect(component).toMatchSnapshot();  
  });
})

describe('SplashComponent', ()=>{
  it('renders without crashing', () => {
  const component = shallow(<Splash/>);
  expect(component).toMatchSnapshot();  
  });
})

describe('SearchComponent', ()=>{
  it('renders without crashing', () => {
  const component = shallow(<Provider store={store}><MemoryRouter><Search/></MemoryRouter></Provider>);
  expect(component).toMatchSnapshot();  
  });
})

describe('App', ()=>{
  it('renders without crashing', () => {
  const component = shallow(<App/>);
  expect(component).toMatchSnapshot();  
  });
})

describe('CreatePostComponent', ()=>{
  it('renders without crashing', () => {
  const component = shallow(<Provider store={store}><MemoryRouter><CreatePost/></MemoryRouter></Provider>);
  expect(component).toMatchSnapshot();  
  });
})

describe('PostComponent', ()=>{
  it('renders without crashing', () => {
  const component = shallow(<Provider store={store}><MemoryRouter><Post post = {{userName:'',displayName:'',displayImg:''}} /></MemoryRouter></Provider>);
  expect(component).toMatchSnapshot();  
  });
})

describe('SettingsComponent', ()=>{
  it('renders without crashing', () => {
  const component = shallow(<Provider store={store}><MemoryRouter><SettingsComponent/></MemoryRouter></Provider>);
  expect(component).toMatchSnapshot();  
  });
})

describe('UserComponent', ()=>{
  it('renders without crashing', () => {
  const component = shallow(<User/>);
  expect(component).toMatchSnapshot();  
  });
})

describe('UserListCoponent', ()=>{
  it('renders without crashing', () => {
  const component = shallow(<UserList users = {[]}/>);
  expect(component).toMatchSnapshot();  
  });
})

/*
describe('NavigationBar', ()=>{
  it('renders without crashing', () => {
  const component = shallow(<Provider store={store}><MemoryRouter><NavigationBar/></MemoryRouter></Provider>);
  expect(component).toMatchSnapshot();  
  });
})

describe('Feed', ()=>{
  it('renders without crashing', () => {
    
  const component = shallow(<Feed posts = {[]}/>);

  expect(component).toMatchSnapshot();  
  });
})


/*


  it("should render title", () => {
    const wrapper = render(
      <MemoryRouter>
        <SignUpComponent />
     </MemoryRouter>
    );
    expect(wrapper.queryAllByText("Bohemian Grove")).toHaveLength(1);
  });

  it("should render title", () => {
    const wrapper = render(
      <MemoryRouter>
        <SignInComponent />
     </MemoryRouter>
    );
    expect(wrapper.queryAllByText("Bohemian Grove")).toHaveLength(1);
  });


it('has a languageName of "TypeScript"', () => {
  const state: typeof initialState = { 
    
    user: {
        userName: ' ',
        displayName: ' ',
        email: ' ',
        profileImg: 'png',
        loginToken: 'TypeScript',
    }
  };
  expect(state.user?.loginToken).toEqual('TypeScript');
});


it('has a userName', () => {
  const state: typeof initialState = { 
    
    user: {
        userName: 'johncena',
        displayName: ' ',
        email: ' ',
        profileImg: 'png',
        loginToken: 'TypeScript',
    }
  };
  expect(state.user?.userName).toEqual('johncena');
});

/*
test('should return the initial state', () => {
  expect(reducer(undefined, {})).toEqual(
    {
      loginToken: " ",
      user:{
        displayName: ' ',
        email: ' ',
        password: ' ',
        profileImg:'png',
        userName:' '
        }
    }
  )
})
*/
/*
describe('manage profile page test suite', () => {
  it('snapshot test', () => {
      const setRouteLeaveHook =jest.fn();

      let wrapper = shallow(
          <Feed params={{router: 
      setRouteLeaveHook}}/>
      );
    expect(wrapper).toMatchSnapshot();
  })
 })
 
 it("should render title", () => {
  const wrapper = render(
    <MemoryRouter>
      <Feed posts = {[]}/>
    </MemoryRouter>
    
  );
  expect(wrapper.queryAllByText("What's on your mind?")).toHaveLength(1);
});


/*
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<MemoryRouter><SignUpComponent /></MemoryRouter>, div);
});

  it("should render title", () => {
    const wrapper = render(
      <MemoryRouter>
        <SignUpComponent />
     </MemoryRouter>
    );
    expect(wrapper.queryAllByText("Bohemian Grove")).toHaveLength(1);
  });

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<MemoryRouter><SignInComponent /></MemoryRouter>, div);
});

  it("should render title", () => {
    const wrapper = render(
      <MemoryRouter>
        <SignInComponent />
     </MemoryRouter>
    );
    expect(wrapper.queryAllByText("Bohemian Grove")).toHaveLength(1);
  });


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<MemoryRouter><Feed /></MemoryRouter>, div);
});

  it("should render title", () => {
    const wrapper = render(
      <MemoryRouter>
        <Feed/>
     </MemoryRouter>
    );
    expect(wrapper.queryAllByText("new-card-title")).toHaveLength(1);
  });


it('has a languageName of "TypeScript"', () => {
  const state: typeof initialState = { 
    loginToken: 'TypeScript',
    user: {
        userName: ' ',
        displayName: ' ',
        email: ' ',
        profileImg: 'png'
    }
  };
  expect(state.loginToken).toEqual('TypeScript');
});
it('has a userName', () => {
  const state: typeof initialState = { 
    loginToken: 'TypeScript',
    user: {
        userName: 'johncena',
        displayName: ' ',
        email: ' ',
        profileImg: 'png'
    }
  };
  expect(state.user.userName).toEqual('johncena');
});


test('should return the initial state', () => {
  expect(reducer(undefined, {})).toEqual([
    {
      username: ' ',
      password: ' '
    }
  ])
})

describe('SignInComponent', () => {
  it("should update the route on click of the button",() => {
    const history = createMemoryHistory();
    const wrapper = render(
      <Router history="{history}">
        <SignInComponent/>
     </Router>
    );
    wrapper.getByText("submit-button").click();
    expect(history.location.pathname).toBe("/feed");
  });
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Router> 
                    <SignInComponent />
                  </Router>, div);
});


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<NavigationBar />, div);
});

describe('manage profile page test suite', () => {


  it('snapshot test', () => {

      const setRouteLeaveHook =jest.fn();

      let wrapper = shallow(
          <Feed params={{router: 
      setRouteLeaveHook}}/>
      );
    expect(wrapper).toMatchSnapshot();
  })
 }) 

 it("should render title", () => {
  const wrapper = render(
    <MemoryRouter>
      <Feed/>
    </MemoryRouter>
    
  );
  expect(wrapper.queryAllByText("What's on your mind?")).toHaveLength(1);
});

 it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
  <MemoryRouter>
    <Feed />
  </MemoryRouter>
  , div);
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<SignUpComponent />, div);
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<SettingsComponent />, div);
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ProfileComponent />, div);
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<CardContainer />, div);
});
*/