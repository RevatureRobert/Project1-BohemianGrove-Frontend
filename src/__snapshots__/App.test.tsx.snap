// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `ShallowWrapper {}`;

exports[`CreatePostComponent renders without crashing 1`] = `ShallowWrapper {}`;

exports[`Feed renders without crashing 1`] = `ShallowWrapper {}`;

exports[`GlobalFeedComponent renders without crashing 1`] = `ShallowWrapper {}`;

exports[`NavigationBar renders without crashing 1`] = `ShallowWrapper {}`;

exports[`PostComponent renders without crashing 1`] = `ShallowWrapper {}`;

exports[`ProfileComponent renders without crashing 1`] = `ShallowWrapper {}`;

exports[`SearchComponent renders without crashing 1`] = `ShallowWrapper {}`;

exports[`SettingsComponent renders without crashing 1`] = `ReactWrapper {}`;

exports[`SettingsComponent renders without crashing 2`] = `ShallowWrapper {}`;

exports[`SignInUpComponent renders without crashing 1`] = `ShallowWrapper {}`;

exports[`SignUpComponent renders without crashing 1`] = `ShallowWrapper {}`;

exports[`SplashComponent renders without crashing 1`] = `ShallowWrapper {}`;

exports[`UserComponent renders without crashing 1`] = `ShallowWrapper {}`;

exports[`UserListCoponent renders without crashing 1`] = `ShallowWrapper {}`;
